False positive in RightCurly with 'alone_or_singleline' RightCurlyOption in try/catch

I have read check documentation: https://checkstyle.org/checks/blocks/rightcurly.html#RightCurly

Related documentation: https://checkstyle.org/property_types.html#RightCurlyOption

From `alone_or_singleline`:
> The brace must be alone on the line, yet single-line format of block is allowed. 

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac RCurlyTest.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <property name="tabWidth" value="4"/>
    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="RightCurly">
            <property name="option" value="alone_or_singleline"/>
        </module>
    </module>
</module>

/var/tmp $ cat RCurlyTest.java
public class RCurlyTest {
	public static void main(String args[]) {
		try { System.out.println(args[0]); } catch (Exception e) {}
	}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.3.1-all.jar -c config.xml RCurlyTest.java
Starting audit...
[ERROR] [...]/RCurlyTest.java:3:44: '}' at column 38 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

Since the whole try..catch construct is in a single line, I expect the RightCurly test to pass (i.e. no errors raised).

As a side-note. Column number in the error message seems to be wrong when tabs are used in the source file. In the above example you can see that:

 * RCurlyTest.java:3:44 is correct and honours the top level tabWidth property in the config.xml file
 * However the message incorrectly refers to "column 38"

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java
@@ -328,9 +328,16 @@ public class RightCurlyCheck extends AbstractCheck {
             nextToken = Details.getNextToken(nextToken);
         }
 
-        if (nextToken != null && nextToken.getType() == TokenTypes.DO_WHILE) {
-            final DetailAST doWhileSemi = nextToken.getParent();
-            nextToken = Details.getNextToken(doWhileSemi);
+        // sibling tokens should be allowed on a single line
+        final int[] tokensWithBlockSibling = {
+            TokenTypes.DO_WHILE,
+            TokenTypes.LITERAL_FINALLY,
+            TokenTypes.LITERAL_CATCH,
+        };
+
+        if (TokenUtil.isOfType(nextToken, tokensWithBlockSibling)) {
+            final DetailAST parent = nextToken.getParent();
+            nextToken = Details.getNextToken(parent);
         }
 
         return TokenUtil.areOnSameLine(details.lcurly, details.rcurly)

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheckTest.java
@@ -307,6 +307,13 @@ public class RightCurlyCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testAloneOrSingleLineTryCatchBlock() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWithInlineConfigParser(
+                getPath("InputRightCurlyTestAloneOrSinglelineTryCatchBlock.java"), expected);
+    }
+
+    @Test
     public void testCatchWithoutFinally() throws Exception {
         final String[] expected = {
             "19:9: " + getCheckMessage(MSG_KEY_LINE_SAME, "}", 9),
@@ -387,9 +394,8 @@ public class RightCurlyCheckTest extends AbstractModuleTestSupport {
     public void testTryWithResourceAloneSingle() throws Exception {
         final String[] expected = {
             "27:9: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 9),
-            "36:64: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 64),
-            "44:15: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 15),
-            "46:13: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 13),
+            "43:15: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 15),
+            "45:13: " + getCheckMessage(MSG_KEY_LINE_ALONE, "}", 13),
         };
         verifyWithInlineConfigParser(
                 getPath("InputRightCurlyTryWithResourceAloneSingle.java"), expected);

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestAloneOrSinglelineTryCatchBlock.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTestAloneOrSinglelineTryCatchBlock.java
@@ -1,0 +1,30 @@
+/*
+RightCurly
+option = ALONE_OR_SINGLELINE
+tokens = LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+public class InputRightCurlyTestAloneOrSinglelineTryCatchBlock {
+
+     private void foo() {
+
+            try {
+                int i = 5; int b = 10;
+            }
+            catch (Exception e) { }
+     }
+     private void testSingleLineTryBlock() {
+         try { } catch (Exception e) { }
+         try {int x = 5;} catch (RuntimeException e) { } catch (Exception e) { }
+         try { } catch (RuntimeException e) { } catch (Exception e) { } finally { foo();}
+         try (BufferedReader br1 = new BufferedReader(null)) {} catch (IOException e) { ; }
+    }
+
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTryWithResourceAloneSingle.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyTryWithResourceAloneSingle.java
@@ -32,7 +32,6 @@ class InputRightCurlyTryWithResourceAloneSingle {
                 BufferedReader br2 = new BufferedReader(br1)) { ; }
         catch (IOException e) { ; }
         try (BufferedReader br1 = new BufferedReader(null);
-                // violation below ''}' at column 64 should be alone on a line'
                 BufferedReader br2 = new BufferedReader(br1)) {} catch (IOException e) { ; }
         try (BufferedReader br1 = new BufferedReader(null);
                 BufferedReader br2 = new BufferedReader(br1)) {

